x-pack/plugins/rule_registry/server/rule_data_plugin_service/index_info.ts

```diff
@@ -115,6 +115,11 @@ export class IndexInfo {
    *   - e.g. if baseName='.alerts-observability', include '.alerts-observability.apm'
    *
    * @example '.alerts-security.alerts-default*', '.alerts-security.alerts*'
+   *
+   * Warning: Reading alerts from this pattern has a side affect that some of them
+   * might come from other spaces with the same prefix.
+   * @example '.alerts-security.alerts-test' and '.alerts-security.alerts-test-2'
+   *
    */
   public getPatternForReading(namespace?: string): string {
     return `${joinWithDash(this.baseName, namespace)}*`;
```

Comment:
I wonder if here we can use either the namespace or `*` if a namespace is not provided? I think that could keep the `getReader` API simpler.
```suggestion
    return `${joinWithDash(this.baseName, namespace ?? '*')}*`;
```

Reply:
Yeah, just would like to make sure there's no situation that needs both namespace and wildcard.